searx/engines/__init__.py

```diff
@@ -67,61 +53,57 @@ def load_engine(engine_data):
         engine_name = engine_name.lower()
         engine_data['name'] = engine_name
 
+    # load_module
     engine_module = engine_data['engine']
-
     try:
         engine = load_module(engine_module + '.py', engine_dir)
     except (SyntaxError, KeyboardInterrupt, SystemExit, SystemError, ImportError, RuntimeError):
         logger.exception('Fatal exception in engine "{}"'.format(engine_module))
         sys.exit(1)
-    except:
+    except BaseException:
         logger.exception('Cannot load engine "{}"'.format(engine_module))
         return None
 
+    # set engine attributes from engine_data
     for param_name, param_value in engine_data.items():
-        if param_name == 'engine':
-            pass
-        elif param_name == 'categories':
+        if param_name == 'categories':
             if param_value == 'none':
                 engine.categories = []
             else:
                 engine.categories = list(map(str.strip, param_value.split(',')))
-        else:
+        elif param_name != 'engine':
             setattr(engine, param_name, param_value)
 
+    # set default attributes
     for arg_name, arg_value in engine_default_args.items():
         if not hasattr(engine, arg_name):
-            setattr(engine, arg_name, arg_value)
+            setattr(engine, arg_name, copy.deepcopy(arg_value))
+
+    # check if engine is inactive
+    if getattr(engine, 'inactive', False) is True:
+        return None
 
     # checking required variables
     for engine_attr in dir(engine):
         if engine_attr.startswith('_'):
             continue
-        if engine_attr == 'inactive' and getattr(engine, engine_attr) is True:
-            return None
         if getattr(engine, engine_attr) is None:
             logger.error('Missing engine config attribute: "{0}.{1}"'
                          .format(engine.name, engine_attr))
             sys.exit(1)
 
     # assign supported languages from json file
-    if engine_data['name'] in ENGINES_LANGUAGES:
-        setattr(engine, 'supported_languages', ENGINES_LANGUAGES[engine_data['name']])
+    if engine_name in ENGINES_LANGUAGES:
+        setattr(engine, 'supported_languages', ENGINES_LANGUAGES[engine_name])
 
     # find custom aliases for non standard language codes
-    if hasattr(engine, 'supported_languages'):
```

Comment:
The condition is removed because it is always True: `engine_default_args` contains `supported_languages`, so the ` for arg_name, arg_value in engine_default_args.items():...` set the attribute `supported_languages`.